### DISCUSSION TITLE
Is `ShortcutHelper` really needed given `Command/KeyBindings`?

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#2974] [Q&A]
@BDisp, I am studying the code in `ViewKeyboard.cs` and looking at `Shortcut` etc...

It seems to me this logic is duplicative with `Command/KeyBindings`.  

For example, if `Menu` was modified to create `Command`s and call `AddKeyBinding` for each menu item's `shortcut` we could get rid of all this code? The other places where `ShortcutHelper` is used are:

## `TextView` - It's only using `ShortcutHelper.GetModifiersKey()` and in a weird way:
```cs
var result = InvokeKeyBindings (new (ShortcutHelper.GetModifiersKey (a),
    new KeyModifiers () { Alt = a.IsAlt, Ctrl = a.IsCtrl, Shift = a.IsShift }));
if (result != null)
	return (bool)result;
```

This could be easily addressed.

## Scenarios (DyanmicMenuBar, VkPacketSimulator, Editor) could all be modified to just use KeyBindings.

Do you have thoughts or suggestions on this?

**Comment by BDisp** [comment]
If both have identical feature that can be addressed by the `Command/KeyBindings` I agree. The later come after the `ShortcutHelper` but is much more important than the prior. Good point.

**Reply by tig** [reply]
Step one will be to refactor Menu to use Command/KeyBindings. 

This will prove that it's possible given how complex Menu is. 

When i get a chance I'll open an issue "Refactor Menu to use Command/KeyBindings".

**Comment by BDisp** [comment]
> Do you have thoughts or suggestions on this?

For now no but I can help in found a way.
